akka-actor/src/main/scala/akka/actor/dungeon/Dispatch.scala

```diff
@@ -147,10 +155,14 @@ private[akka] trait Dispatch { this: ActorCell ⇒
     unwrappedMessage match {
       case _: NoSerializationVerificationNeeded ⇒ envelope
       case msg ⇒
-        val deserializedMsg = serializeAndDeserializePayload(msg)
-        envelope.message match {
-          case dl: DeadLetter ⇒ envelope.copy(message = dl.copy(message = deserializedMsg))
-          case _              ⇒ envelope.copy(message = deserializedMsg)
+        Try(serializeAndDeserializePayload(msg)) match {
```

Comment:
Would have preferred a raw try catch I guess, since we're not gaining much with using the `Try/immediately-match`.